docs/src/main/asciidoc/reactive-postgres-client.adoc

```diff
@@ -0,0 +1,450 @@
+
+IMPORTANT: If you are not familiar with the {project-name} Vert.x extension, consider reading the link:using-vertx.html[Using Eclipse Vert.x] guide first.
+
+The application shall manage fruit entities:
+
+[source,java]
+----
+public class Fruit {
+
+    public Long id;
+
+    public String name;
+
+    public Fruit() {
+    }
+
+    public Fruit(String name) {
+        this.name = name;
+    }
+
+    public Fruit(Long id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+}
+----
+
+[TIP]
+====
+Do you need a ready-to-use Postgres server to try out the examples?
+
+[source,shell]
+----
+docker run --ulimit memlock=-1:-1 -it --rm=true --memory-swappiness=0 --name quarkus_test -e POSTGRES_USER=quarkus_test -e POSTGRES_PASSWORD=quarkus_test -e POSTGRES_DB=quarkus_test -p 5432:5432 postgres:10.5
+----
+====
+
+== Installing
+
+=== Reactive Postgres Client extension
+
+First, make sure your project has the `quarkus-reactive-pg-client` extension enabled.
+If you are creating a new project, set the `extensions` parameter as follows:
+
+[source, shell, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=reactive-pg-client-quickstart \
+    -Dextensions="quarkus-reactive-pg-client"
+----
+
+If you have an already created project, the `quarkus-reactive-pg-client` extension can be added to an existing {project-name} project with the `add-extension` command:
+
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions="quarkus-reactive-pg-client"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-reactive-pg-client</artifactId>
+</dependency>
+----
+
+NOTE: In this guide, we will use the Axle API of the Reactive Postgres Client.
+Read the link:using-vertx.html[Using Eclipse Vert.x] guide to understand the differences between the callback, RxJava and Axle based APIs.
+
+=== JSON Binding
+
+We will expose `Fruit` instances over HTTP in the JSON format.
+Consequently, you also need to add the `quarkus-resteasy-jsonb` extension:
+
+[source,shell]
+----
+mvn quarkus:add-extension -Dextensions="quarkus-resteasy-jsonb"
+----
+
+If you prefer not to use the command line, manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-resteasy-jsonb</artifactId>
+</dependency>
+----
+
+Of course, this is only a requirement for this guide, not any application using the Reactive Postgres Client.
+
+== Configuring
+
+The Reactive Postgres Client can be configured with standard Quarkus datasource properties:
+
+[source]
+.src/main/resources/application.properties
+----
+quarkus.datasource.url=vertx-reactive:postgresql:///quarkus_test
```

Comment:
Really 4 slashes or is this asciidoc formatting escaping?

Reply:
No, 3 slashes. Two are from the prefix `postgresl://` prefix, the last one separates the database from the `host:port` info that is empty in this example. If you prefer I can hardcode the defaults.